ts/screens/wallet/payment/ScanQrCodeScreen.tsx

```diff
@@ -86,92 +76,80 @@ const styles = StyleSheet.create({
     backgroundColor: "transparent",
     height: (screenWidth * 4) / 3,
     width: screenWidth
-  },
-
-  rectangleContainer: {
-    flex: 1,
-    alignItems: "center",
-    justifyContent: "center",
-    backgroundColor: "transparent"
-  },
-
-  rectangle: {
-    height: screenWidth / 2,
-    width: screenWidth / 2,
-    borderWidth: 0,
-    backgroundColor: "transparent"
-  },
-
-  smallBorded: {
-    height: screenWidth / 6,
-    width: screenWidth / 6,
-    borderColor: variables.brandPrimaryInverted,
-    backgroundColor: "transparent",
-    position: "absolute"
-  },
-
-  topRightCorner: {
-    borderTopWidth: 2,
-    borderRightWidth: 2,
-    top: 0,
-    right: 0
-  },
-
-  topLeftCorner: {
-    borderTopWidth: 2,
-    borderLeftWidth: 2,
-    top: 0,
-    left: 0
-  },
-
-  bottomLeftCorner: {
-    borderBottomWidth: 2,
-    borderLeftWidth: 2,
-    bottom: 0,
-    left: 0
-  },
-
-  bottomRightCorner: {
-    borderBottomWidth: 2,
-    borderRightWidth: 2,
-    bottom: 0,
-    right: 0
   }
 });
 
-const QRCODE_SCANNER_REACTIVATION_TIME_MS = 2000;
+/**
+ * Delay for reactivating the QR scanner after a scan
+ */
+const QRCODE_SCANNER_REACTIVATION_TIME_MS = 1000;
 
-const rptIdFromQrCodeString = (qrCodeString: string): Either<t.Errors, RptId> =>
-  PaymentNoticeQrCodeFromString.decode(qrCodeString).chain(
-    rptIdFromPaymentNoticeQrCode
-  );
+/**
+ * Duration of the vibration on valid scan
+ */
+const QRCODE_VALID_VIBRATION_DURATION_MS = 400;
+
+class ScanQrCodeScreen extends React.PureComponent<Props, State> {
+  private scannerReactivateTimeoutHandler: number | undefined;
+
+  /**
+   * Handles valid PagoPA QR codes
+   */
+  private onValidQrCode = (data: ITuple2<RptId, AmountInEuroCents>) => {
```

Comment:
I'm pretty sure the [QrCodeScanner](https://github.com/moaazsidat/react-native-qrcode-scanner) component already vibrates upon scanning a valid qr code. I think the desired behavior here is that the vibration only occurs upon reading a valid PagoPA component, so we should disable the vibration from the component (`vibrate=false`) and have the vibration happen here.

Reply:
There's no way to disable the vibration (at least I don't see the prop), anyway since it already vibrates, I've removed this code

Reply:
You probably need to update `react-native-qrcode-scanner`'s type file (in `@types`). Since there's no official version, I added it. This `vibrate` property was added after I created the type file, so it's not there (`vibrate?: boolean`). [Here](https://github.com/moaazsidat/react-native-qrcode-scanner/blob/94ab4db17b6a28af28c4f6d0886e4cabdffa3b99/index.js#L164) is the code that handles `vibrate`.